addons/hr_timesheet/models/project.py

```diff
@@ -129,7 +129,7 @@ def _compute_analytic_account_active(self):
     @api.depends('timesheet_ids.unit_amount')
     def _compute_effective_hours(self):
         for task in self:
-            task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')), 2)
+            task.effective_hours = round(sum(self.env['account.analytic.line'].sudo().search([('task_id', '=', task.id)]).mapped('unit_amount')), 2)
```

Comment:
What about this diff?
```diff

diff --git a/addons/hr_timesheet/models/project.py b/addons/hr_timesheet/models/project.py
index 554948b69f5..e8c94bc62b3 100644
--- a/addons/hr_timesheet/models/project.py
diff --git a/addons/hr_timesheet/models/project.py b/addons/hr_timesheet/models/project.py
index 554948b69f5..86496c18f4e 100644
--- a/addons/hr_timesheet/models/project.py
+++ b/addons/hr_timesheet/models/project.py
@@ -128,8 +128,22 @@ class Task(models.Model):

     @api.depends('timesheet_ids.unit_amount')
     def _compute_effective_hours(self):
-        for task in self:
-            task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')), 2)
+        # We need to ensure that all the users, whatever their right access is, get the same result when saving.
+        # On the other hand, while the behavior during edition is that onchange are triggerd on the cache records
+        # (which are the ones that the user actually has access to), the calculation is made on what the user actually
+        # sees. Bellow code ensures that the correct value is computed at save time while allowing the onchange to take
+        # the change made by the user into account. This is of course not perfect, but the best we can do for now.
+        if all(self._ids):
+            timesheet_unit_amount_per_task = self.env['account.analytic.line'].read_group(
+                [('task_id', 'in', self.ids)], ['task_id', 'unit_amount'], ['task_id', 'unit_amount']
+            )
+            timesheet_unit_amount_per_task = {record['task_id'][0]: record['unit_amount']
+                                              for record in timesheet_unit_amount_per_task}
+            for task in self:
+                task.effective_hours = timesheet_unit_amount_per_task.get(task.id, 0.0)
+        else:
+            for task in self:
+                task.effective_hours = round(sum(task.timesheet_ids.mapped('unit_amount')), 2)

     @api.depends('effective_hours', 'subtask_effective_hours', 'planned_hours')
     def _compute_progress_hours(self):
```

It is not perfect either... As the onchange won't be correct, but I can't find any solution to it.

@xavierbol , @tivisse can you have a look?

Reply:
@IT-Ideas we already do something like that in master so LGTM. 🙂 

https://github.com/odoo/odoo/blob/23220de37ee2493b1eb2be6a826f28453fb3b43c/addons/hr_timesheet/models/project.py#L280-L289

Reply:
@xavierbol Thank you for checking! I just tested in master and it indeed works well. I will push a new fix which aligns with the master version then :smile: @IT-Ideas what do you think?